src/m365/outlook/commands/message/message-remove.ts

```diff
@@ -0,0 +1,142 @@
+import auth from '../../../../Auth.js';
+import { Logger } from '../../../../cli/Logger.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { accessToken } from '../../../../utils/accessToken.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import { cli } from '../../../../cli/cli.js';
+import { validation } from '../../../../utils/validation.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  userId?: string;
+  userName?: string;
+  force?: boolean
+}
+
+class OutlookMessageRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MESSAGE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes a specifc message from a mailbox';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const isAppOnlyAccessToken: boolean | undefined = accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken);
+    let requestUrl = '';
+
+    if (isAppOnlyAccessToken) {
+      if (!args.options.userId && !args.options.userName) {
+        throw `The option 'userId' or 'userName' is required when removing a message using application permissions`;
+      }
+
+      if (args.options.userId && args.options.userName) {
+        throw `Both options 'userId' and 'userName' cannot be set when removing a message using application permissions`;
+      }
+
+      if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+        throw `The value '${args.options.userId}' for 'userId' option is not a valid GUID`;
+      }
+
+      if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+        throw `The value '${args.options.userName}' for 'userName' option is not a valid user principal name`;
+      }
```

Comment:
We can remove these validators here. Like mentioned before, this are values that we can check before executing the command in a #initValidators function.